data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -429,7 +431,9 @@ private Object setChildrenAndTriggerPostLoad(Object instance, MappingContext<?>
                             } else {
                                 v = readProperty(rs, ctx, prop);
                                 if (v == null) {
-                                    if (!prop.isOptional() && !nullableEmbedded) {
+                                    if (ctx.persistentEntity.findAnnotation(Embeddable.class).isPresent() || ctx.persistentEntity.findAnnotation(EmbeddedId.class).isPresent()) {
```

Comment:
yes `ctx.persistentEntity.hasAnnotation(Embeddable.class)` it will be more efficient

Reply:
`hasAnnotation()` does not appear to exist? only get and find, which all do the same work. `MethodInvocationContext` has `hasAnnotation` but not `MappingContext`

Reply:
call `getAnnotationMetadata().hasAnnotation`

Reply:
ah righto, done